src/back-end/lib/db/opportunity/sprint-with-us.ts

```diff
@@ -1671,46 +1676,116 @@ export const submitIndividualQuestionEvaluations = tryDb<
 >(async (connection, id, evaluationParams, session) => {
   const now = new Date();
   await connection.transaction(async (trx) => {
-    for (const evaluationId of evaluationParams.evaluations) {
-      const [statusRecord] =
-        await connection<SWUTeamQuestionResponseEvaluationStatusRecord>(
-          "swuTeamQuestionResponseEvaluationStatuses"
-        )
-          .transacting(trx)
-          .insert(
-            {
-              id: generateUuid(),
-              teamQuestionResponseEvaluation: evaluationId,
-              createdAt: now,
-              createdBy: session.user.id,
-              status: SWUTeamQuestionResponseEvaluationStatus.Submitted,
-              note: evaluationParams.note
-            },
-            "*"
-          );
+    await Promise.all(
+      evaluationParams.evaluations.map(
+        async ({ id: evaluationId, proposal: { id: proposalId } }) => {
+          const [statusRecord] =
+            await connection<SWUTeamQuestionResponseEvaluationStatusRecord>(
+              "swuTeamQuestionResponseEvaluationStatuses"
+            )
+              .transacting(trx)
+              .insert(
+                {
+                  id: generateUuid(),
+                  teamQuestionResponseEvaluation: evaluationId,
+                  createdAt: now,
+                  createdBy: session.user.id,
+                  status: SWUTeamQuestionResponseEvaluationStatus.Submitted,
+                  note: evaluationParams.note
+                },
+                "*"
+              );
 
-      // Update evaluation root record
-      await connection<RawSWUTeamQuestionResponseEvaluation>(
-        "swuTeamQuestionResponseEvaluations"
+          // Update evaluation root record
+          await connection<RawSWUTeamQuestionResponseEvaluation>(
+            "swuTeamQuestionResponseEvaluations"
+          )
+            .transacting(trx)
+            .where({ id })
+            .update(
+              {
+                updatedAt: now
+              },
+              "*"
+            );
+
+          if (!statusRecord) {
+            throw new Error("unable to update team question evaluation");
+          }
+
+          // Check if proposal can be moved into question consensus
+          const [
```

Comment:
I think this makes sense, although you might consider extracting into a separate function (the proposal check piece).